Blogger.API/Infrastructure/Respositories/BlogRepository.cs

```diff
@@ -25,17 +25,17 @@ public async Task CreateAsync(Blog blog)
 
         public async Task<List<Blog>> GetAllAsync()
         {
-            return await _dbContext.Blogs.ToListAsync();
+            return await _dbContext.Blogs.Where(b => b.IsDeleted == false).ToListAsync();
         }
 
         public Task<Blog> GetByIdAsync(Guid id)
         {
-            return _dbContext.Blogs.SingleOrDefaultAsync(b => b.Id == id);
+            return _dbContext.Blogs.Where(b => b.IsDeleted == false).SingleOrDefaultAsync(b => b.Id == id);
```

Comment:
no need for where clause. You can do: 

)dbContext.Blogs.SingleOrDefaultAsync(b => b.Id == id && !b.IsDeleted)